### PR TITLE
Allow ReleaseGoldInstance beneficiary to transfer beneficiary

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR allows a ReleaseGoldInstance beneficiary to transfer the beneficiary to another address

### Tested

- Unit tests

### CODE REVIEW

File: packages/protocol/test/governance/release_gold.ts [hunk 1]
```diff
@@ -377,6 +380,44 @@ contract('ReleaseGold', (accounts: string[]) => {
...
+      releaseGoldInstance = await ReleaseGoldInstance.at(releaseGoldInstanceAddress)
+    })
+
+    it('should set a new beneficiary as the old beneficiary', async () => {
+      await releaseGoldInstance.setBeneficiary(newBeneficiary, { from: beneficiary })
+      const actualBeneficiary = await releaseGoldInstance.beneficiary()
+      assertSameAddress(actualBeneficiary, newBeneficiary)
+    })
+
+    it('should revert when setting a new beneficiary from the releas owner', async () => {
```
Comment (asaj) [inline packages/protocol/test/governance/release_gold.ts]: typo

File: packages/protocol/contracts/governance/ReleaseGoldInstance.sol [hunk 2]
```diff
@@ -198,6 +199,7 @@ contract ReleaseGoldInstance is UsingRegistry, ReentrancyGuard, IReleaseGoldInst
 
     setRegistry(registryAddress);
     beneficiary = _beneficiary;
+    emit BeneficiarySet(beneficiary);
```
Comment (asaj) [inline packages/protocol/contracts/governance/ReleaseGoldInstance.sol]: Please define a `_setBeneficiary` that doesn't have the modifier, and then use that here, and then `setBeneficiary` just wraps `_setBeneficiary` with the modifier

**Review by asaj — APPROVED** [review]

**Comment by lucasege** [comment]
How will this play in with `ReleaseGold` being deployed under a 2-of-2 multisig (one address is the beneficiary, one is the grant-ing entity)? Seems to me that the beneficiary multisig address will need to be swapped as well, which will probably require `ReleaseGold` to have a reference to the `Multisig`.
Maybe there are other solutions (maybe the multisig can listen for the `setBeneficiary` event some how)?
Any ideas?

**Comment by nambrot** [comment]
I think the only reasonable way will be to have the beneficiary be responsible to rotate themselves with the help of the release manager. Ideally we could have a modified multisig that reads one of the owners from `ReleaseGold`, but it's not worth it the security risk IMO

**Comment by asaj** [comment]
> I think the only reasonable way will be to have the beneficiary be responsible to rotate themselves with the help of the release manager. Ideally we could have a modified multisig that reads one of the owners from ReleaseGold, but it's not worth it the security risk IMO


In that case, to avoid the beneficiary accidentally swapping addresses without updating the multisig, should we require `setBeneficiary` to be `onlyMultisig`?

**Comment by nambrot** [comment]
Great idea, @lucasege would you mind making the change once this goes in?

**Comment by lucasege** [comment]
> Great idea, @lucasege would you mind making the change once this goes in?

Yeah can do, if it gets automerged in a reasonable amount of time

File: packages/protocol/contracts/governance/ReleaseGoldInstance.sol [hunk 3]
```diff
@@ -254,6 +256,16 @@ contract ReleaseGoldInstance is UsingRegistry, ReentrancyGuard, IReleaseGoldInst
     emit DistributionLimitSet(beneficiary, maxDistribution);
   }
 
+  /**
+   * @notice Sets the beneficiary of the instance
+   * @param newBeneficiary The address of the new beneficiary
+   */
+  function setBeneficiary(address payable newBeneficiary) external onlyBeneficiary {
```
Comment (codyborn) [inline packages/protocol/contracts/governance/ReleaseGoldInstance.sol:262]: Similarly, should we allow updating of the refundAddress?

Comment (codyborn) [inline packages/protocol/contracts/governance/ReleaseGoldInstance.sol:262]: Do we have tests for this contract?

Comment (nambrot) [reply packages/protocol/contracts/governance/ReleaseGoldInstance.sol:262]: Ideally, probably yes, though the likelihood of key rotation for that refund address seem to be much lower @asaj

Comment (nambrot) [reply packages/protocol/contracts/governance/ReleaseGoldInstance.sol:262]: Yes, does it not show in the diff for you?

Comment (codyborn) [reply packages/protocol/contracts/governance/ReleaseGoldInstance.sol:262]: I see it, thanks!

**Review by codyborn — APPROVED** [review]

**Pull request merged**